pkg/activator/handler/context_handler.go

```diff
@@ -52,6 +54,16 @@ type contextHandler struct {
 func (h *contextHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	namespace := r.Header.Get(activator.RevisionHeaderNamespace)
 	name := r.Header.Get(activator.RevisionHeaderName)
+
+	// If the headers aren't explicitly specified, then decode the revision
+	// name and namespace from the Host header.
+	if name == "" || namespace == "" {
+		parts := strings.SplitN(r.Host, ".", 3)
+		if len(parts) == 3 && parts[2] == "svc."+network.GetClusterDomainName() {
```

Comment:
splitting in 4 and checking [3] == svc and then [4] == getclusterdomain name is faster, so we should do that, since this might happen on every request

https://gist.github.com/vagababov/8d2a2e9bd30e7c23ef3ea5651f43353c

Reply:
done